ios/RNViewShot.m

```diff
@@ -88,7 +101,17 @@ - (dispatch_queue_t)methodQueue
       scrollView.contentOffset = CGPointZero;
       scrollView.frame = CGRectMake(0, 0, scrollView.contentSize.width, scrollView.contentSize.height);
     }
-    UIGraphicsBeginImageContextWithOptions(size, NO, 0);
+
+    if ([target intValue] == -1) {
```

Comment:
(sorry to not have noticed this before) i'm not sure about this `if()` block at all, I think we should stick with `UIGraphicsBeginImageContextWithOptions(size, NO, 0);`:
the idea is to let the user potentially resize the view.

in case of snapshotting the window, I'm pretty sure that `UIGraphicsBeginImageContextWithOptions(view.window.bounds.size, NO, [UIScreen mainScreen].scale)` is already equivalent to `UIGraphicsBeginImageContextWithOptions(size, NO, 0)`

according to [this doc](https://developer.apple.com/documentation/uikit/1623912-uigraphicsbeginimagecontextwitho?language=objc), setting `0` to scale option defaults to the device pixel ratio already so no need to do `[UIScreen mainScreen].scale` ?

BTW it would be consistent with Android I think?

Reply:
yeah , probably need to branch at this line

https://github.com/foxmicha/react-native-view-shot/blob/8ca87ce03d8f98dbef1315109672ea8a9007ad73/ios/RNViewShot.m#L88

to do `view.window.bounds.size`

**unless `view.window.bounds.size` IS `view.bounds.size`** ? (I don't have good knowledge of this)

Reply:
After doing some experimentation, it appears you are correct. That `if()` statement was added as a means to capture the direct pixel resolution of a scaled device, such as a Retina display. However, this doesn't appear to be absolutely necessary.

As for `view.windows.bounds.size` vs `view.bounds.size` because we are setting the `view = window.rootViewController.view` in the first `if()` check, they appear to be functionally the same. If we hadn't already selected the `rootViewController.view` we'd want want to use the window's bounds that our child view was residing in.